addons/misra.py

```diff
@@ -2763,6 +2763,13 @@ def misra_12_4(self, cfg):
             max_value = (1 << bits) - 1
             if not is_constant_integer_expression(expr):
                 continue
+            if expr.astParent.str == ":" and expr.astParent.astParent.str == "?":
```

Comment:
Do you know that `expr.astParent` and `expr.astParent.astParent` are not None somehow?

This code is unfortunate I have the feeling you will write false positives here:

    x = (0u==0u) ? 0u : (2*(0u-1u));

you could walk up the ast for every expr but that seems unfortunate too.. is it possible to skip the whole unreachable expression?